express-api/src/controllers/projects/projectsController.ts

```diff
@@ -287,7 +259,24 @@ export const searchProjects = async (req: Request, res: Response) => {
  * @returns {Response}      A 200 status with the an array of projects.
  */
 export const filterProjects = async (req: Request, res: Response) => {
-  const projects = await filterProjectsByAgencies(req, res);
+  const filter = ProjectFilterSchema.safeParse(req.query);
+  const includeRelations = req.query.includeRelations === 'true';
+  const kcUser = req.user as unknown as SSOUser;
+  if (!filter.success) {
+    return res.status(400).send('Could not parse filter.');
+  }
+  const filterResult = filter.data;
+
+  let projects;
+  if (isAdmin(kcUser) || isAuditor(kcUser)) {
+    projects = await projectServices.getProjects(filterResult as ProjectFilter, includeRelations);
+  } else {
+    // get array of user's agencies
+    const usersAgencies = await userServices.getAgencies(kcUser.preferred_username);
+    filterResult.agencyId = usersAgencies;
```

Comment:
Doing this does mean that if the user did specify a certain agency they wanted to filter on in the query params, it would be overridden here instead. I imagine it could still be valid for a non-admin user to filter on a specific agency... Like if they wanted to get projects from just a specific child agency under their parent agency for example.

Reply:
That's a good point, If a user tries to filter by an agency that is outside of their own, then yes it overrides it. By default it's always returning the records based on the user's agencies. Right now, a user can still filter the projects by child agency if they belong to a parent agency.